src/plugins/vis_builder/public/application/components/data_tab/index.tsx

```diff
@@ -3,19 +3,265 @@
+      sourceAgg: null,
+    });
+
+    let updatedAggConfigs = aggProps.aggConfigs?.aggs;
+
+    if (combine) {
+      // remove the previously selected Aggreagtion
+      updatedAggConfigs = updatedAggConfigs?.filter((agg) => agg.id !== destinationFieldToCombine);
+    }
+    if (updatedAggConfigs) {
+      dispatch(updateAggConfigParams(updatedAggConfigs.map((agg) => agg.serialize())));
+    }
+  };
+
+  const handleConfigurationPanelTransition = ({
+    source,
+    destination,
+    combine,
+    draggableId,
+  }: DropResult) => {
+    const destinationSchemaName = destination?.droppableId || combine?.droppableId;
+    const destinationAggFields = schemaDisplayStates[destinationSchemaName];
+
+    const sourceAggId = draggableId;
+    const destinationAggId = destinationAggFields[destination?.index] || combine?.draggableId;
+
+    const destinationSchema = schemas.all.find(
+      (schema) => schema.name === (destination?.droppableId || combine?.droppableId)
+    );
+
+    if (!destinationSchema) {
+      // Invalid Transition
+      return;
+    }
+
+    const sourceAgg = aggProps.aggConfigs?.aggs.find((agg) => agg.id === sourceAggId);
+    const sourceFieldName = sourceAgg?.fieldName();
+
+    if (!combine) {
+      if (source?.droppableId === destination?.droppableId && source !== null) {
+        if (source.index === destination.index) {
+          // Moving the same element
+          return;
+        } else {
+          // Reordering of the selections within a same group
+          dispatch(
+            reorderAgg({
+              sourceId: sourceAggId,
+              destinationId: destinationAggId,
+            })
+          );
+        }
+      } else if (
+        source?.droppableId !== destination?.droppableId &&
+        source !== null &&
+        destination !== null
+      ) {
+        // Moving a element from one Dropable Box to another
+
+        const destinationLimit = destinationSchema?.max;
+        if (destinationLimit && destinationAggFields.length <= destinationLimit) {
+          // Case 1: Destination has space
+          // We Need to update sourceAgg
+
+          createNewAggConfig({
+            fieldName: sourceFieldName,
+            sourceGroup: source.droppableId,
+            destinationSchema,
+            aggProps,
+            aggService,
+            sourceAgg,
+          });
+
+          // Remove the sourceAggConfig from the updated Config
+          const updatedAggConfig = aggProps.aggConfigs?.aggs.filter(
+            (agg) => agg.id !== sourceAggId
+          );
+
+          if (updatedAggConfig?.length) {
+            dispatch(updateAggConfigParams(updatedAggConfig.map((agg) => agg.serialize())));
+          }
+        } else {
+          // Case 2 : Destination has no space
+          return;
+        }
+      }
+    } else if (combine !== null) {
+      // Combining Elements
+      // TODO: Do we need to restrict drag and drop features amongst the Dragables in one Droppables
```

Comment:
What do you mean by this?

Reply:
It was a initial idea if we wanted to restrict combining fields across schemas and not within a schema. Based on the current implementation there are no such restrictions and all of the fields can be combined irrespective of the schema they belong to.